superset-frontend/src/SqlLab/components/SqlEditor/index.tsx

```diff
@@ -0,0 +1,742 @@
+  appContainer.getAttribute('data-bootstrap') || '{}',
+);
+const validatorMap =
+  bootstrapData?.common?.conf?.SQL_VALIDATORS_BY_ENGINE || {};
+const scheduledQueriesConf = bootstrapData?.common?.conf?.SCHEDULED_QUERIES;
+
+interface SqlEditorProps {
+  actions: object;
+  database?: DatabaseObject;
+  latestQuery?: Query;
+  tables: any[];
+  editorQueries: QueryEditor[];
+  dataPreviewQueries: object[];
+  queryEditorId: string;
+  hideLeftBar?: boolean;
+  defaultQueryLimit: number;
+  maxRow: number;
+  displayLimit: number;
+  saveQueryWarning?: string;
+  scheduleQueryWarning?: string;
+}
+
+const LimitSelectStyled = styled.span`
+  .ant-dropdown-trigger {
+    align-items: center;
+    color: black;
+    display: flex;
+    font-size: 12px;
+    margin-right: ${({ theme }) => theme.gridUnit * 2}px;
+    text-decoration: none;
+    span {
+      display: inline-block;
+      margin-right: ${({ theme }) => theme.gridUnit * 2}px;
+      &:last-of-type: {
+        margin-right: ${({ theme }) => theme.gridUnit * 4}px;
+      }
+    }
+  }
+`;
+
+const StyledToolbar = styled.div`
+  padding: ${({ theme }) => theme.gridUnit * 2}px;
+  background: ${({ theme }) => theme.colors.grayscale.light5};
+  display: flex;
+  justify-content: space-between;
+  border: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+  border-top: 0;
+
+  form {
+    margin-block-end: 0;
+  }
+
+  .leftItems,
+  .rightItems {
+    display: flex;
+    align-items: center;
+    & > span {
+      margin-right: ${({ theme }) => theme.gridUnit * 2}px;
+      display: inline-block;
+
+      &:last-child {
+        margin-right: 0;
+      }
+    }
+  }
+
+  .limitDropdown {
+    white-space: nowrap;
+  }
+`;
+
+const SqlEditor: FC<SqlEditorProps> = ({
+  actions,
+  database = null,
+  latestQuery,
+  tables,
+  editorQueries,
+  dataPreviewQueries,
+  queryEditorId,
+  hideLeftBar = false,
+  defaultQueryLimit,
+  maxRow,
+  displayLimit,
+  saveQueryWarning,
+  scheduleQueryWarning = null,
+}) => {
+  const theme = useTheme();
+  const dispatch = useDispatch();
+  const queryEditor = useSelector((state: RootState) =>
+    state.sqlLab.queryEditors.find(editor => editor.id === queryEditorId),
+  );
+
+  const [height, setHeight] = useState(0);
+  const [autorun, setAutorun] = useState(queryEditor.autorun);
+  const [ctas, setCtas] = useState('');
+  const [northPercent, setNorthPercent] = useState(queryEditor.northPercent || INITIAL_NORTH_PERCENT);
+  const [southPercent, setSouthPercent] = useState(queryEditor.southPercent || INITIAL_SOUTH_PERCENT);
+  const [sql, setSQL] = useState(queryEditor.sql);
+  const [autocompleteEnabled, setAutocompleteEnabled] = useState(
+    getItem(LocalStorageKeys.sqllab__is_autocomplete_enabled, true),
+  );
+  const [showCreateAsModal, setShowCreateAsModal] = useState(false);
+  const [createAs, setCreateAs] = useState('');
+
+  const sqlEditorRef = useRef(null);
+  const northPaneRef = useRef(null);
+
+  useState(() => {
```

Comment:
Are you sure that you wanted to use `useState` here?

Reply:
This `useState` for `UNSAFE_componentWillMount` replacement